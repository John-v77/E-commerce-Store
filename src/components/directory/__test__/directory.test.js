import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Directory from '../directory.component';

const renderWithRouter = (component) => {
  return render(
    <BrowserRouter>
      {component}
    </BrowserRouter>
  );
};

describe('Directory Component', () => {
  it('renders all category items', () => {
    renderWithRouter(<Directory />);
    
    expect(screen.getByText('hats')).toBeInTheDocument();
    expect(screen.getByText('jackets')).toBeInTheDocument();
    expect(screen.getByText('shoes')).toBeInTheDocument();
    expect(screen.getByText('womens')).toBeInTheDocument();
    expect(screen.getByText('mens')).toBeInTheDocument();
  });

  it('renders Shop Now text for all categories', () => {
    renderWithRouter(<Directory />);
    
    const shopNowElements = screen.getAllByText('Shop Now');
    expect(shopNowElements).toHaveLength(5);
  });

  it('renders correct number of categories', () => {
    renderWithRouter(<Directory />);
    
    const categoryTitles = ['hats', 'jackets', 'shoes', 'womens', 'mens'];
    categoryTitles.forEach(title => {
      expect(screen.getByText(title)).toBeInTheDocument();
    });
  });

  it('displays category headings as h2 elements', () => {
    renderWithRouter(<Directory />);
    
    const headings = screen.getAllByRole('heading', { level: 2 });
    expect(headings).toHaveLength(5);
    
    const expectedTitles = ['hats', 'jackets', 'shoes', 'womens', 'mens'];
    headings.forEach((heading, index) => {
      expect(heading).toHaveTextContent(expectedTitles[index]);
    });
  });

  it('matches snapshot', () => {
    const { container } = renderWithRouter(<Directory />);
    expect(container.firstChild).toMatchSnapshot();
  });
});