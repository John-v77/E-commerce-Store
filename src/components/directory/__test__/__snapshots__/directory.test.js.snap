// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Directory Component matches snapshot 1`] = `
<div
  class="sc-hKMtZM iKOfmT"
>
  <div
    class="sc-dkzDqf booHWC"
  >
    <div
      class="sc-bczRLJ enGQty"
    />
    <div
      class="sc-gsnTZi iXBACK"
    >
      <h2>
        hats
      </h2>
      <p>
        Shop Now
      </p>
    </div>
  </div>
  <div
    class="sc-dkzDqf booHWC"
  >
    <div
      class="sc-bczRLJ hqcIsC"
    />
    <div
      class="sc-gsnTZi iXBACK"
    >
      <h2>
        jackets
      </h2>
      <p>
        Shop Now
      </p>
    </div>
  </div>
  <div
    class="sc-dkzDqf booHWC"
  >
    <div
      class="sc-bczRLJ bYbDlx"
    />
    <div
      class="sc-gsnTZi iXBACK"
    >
      <h2>
        shoes
      </h2>
      <p>
        Shop Now
      </p>
    </div>
  </div>
  <div
    class="sc-dkzDqf booHWC"
  >
    <div
      class="sc-bczRLJ jbvlOZ"
    />
    <div
      class="sc-gsnTZi iXBACK"
    >
      <h2>
        womens
      </h2>
      <p>
        Shop Now
      </p>
    </div>
  </div>
  <div
    class="sc-dkzDqf booHWC"
  >
    <div
      class="sc-bczRLJ jYSXyS"
    />
    <div
      class="sc-gsnTZi iXBACK"
    >
      <h2>
        mens
      </h2>
      <p>
        Shop Now
      </p>
    </div>
  </div>
</div>
`;
